app/controllers/sessions_controller.rb

```diff
@@ -1,19 +1,22 @@
 class SessionsController < ApplicationController
   def new
-    redirect_to home_path if current_user  
+    redirect_to home_path if current_user
   end
-  
+
   def create
     user = User.where(email: params[:email]).first
-    if user && user.authenticate(params[:password])
+    if user && user.authenticate(params[:password]) && user.active
       session[:user_id] = user.id
       redirect_to home_path, notice: "Welcome #{user.full_name}! You are signed in."
+    elsif user && user.active == false
```

Comment:
`user.active == false` it should already be false at this point right. The previous users should have `true`. There should not be room for `nil`.